### PR TITLE
Add the MeasureData class.

### PR DESCRIPTION
**Pull request by isturdy**
Another part of the DeltaProducer--this tracks all aggregations (including potentially multiple histograms) on a measure.

### CODE REVIEW

File: opencensus/stats/internal/measure_data.cc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    histograms_.emplace_back(b.num_buckets());
+  }
+}
+
+void MeasureData::Add(double value) {
+  // Update using the method of provisional means.
+  ++count_;
+  ABSL_ASSERT(count_ > 0 && "Histogram count overflow.");
+  const double new_mean = mean_ + (value - mean_) / count_;
+  sum_of_squared_deviation_ =
```
Comment (g-easy) [inline opencensus/stats/internal/measure_data.cc]: +=

File: opencensus/stats/internal/measure_data_test.cc [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+  BucketBoundaries distribution_buckets = BucketBoundaries::Explicit({0});
+  Distribution distribution =
+      testing::TestUtils::MakeDistribution(&distribution_buckets);
+  EXPECT_DEBUG_DEATH(
+      {
+        data.AddToDistribution(&distribution);
+        EXPECT_THAT(distribution.bucket_counts(), ::testing::ElementsAre(1, 0));
+      },
+      "");
```
Comment (g-easy) [inline opencensus/stats/internal/measure_data_test.cc]: If you ABSL_ASSERT(false && "Reason.") can you EXPECT("Reason")?

**Comment by g-easy** [comment]
I've signed the CLA.

File: opencensus/stats/internal/measure_data_test.cc (hunk 2 above)
Comment (isturdy) [reply opencensus/stats/internal/measure_data_test.cc]: Yes. It even picks up the log on the previous line, so I do not need to duplicate the error message in the log and the assert.

File: opencensus/stats/internal/measure_data.cc (hunk 1 above)
Comment (isturdy) [reply opencensus/stats/internal/measure_data.cc]: Updating sum_of_squared_deviation requires both the new and old mean. I guess it is a bit clearer to save the old mean locally (formerly in the next line count_ is the new count and mean_ the old mean), but I do not think it saves anything.

**Pull request merged**